stackslib/src/net/api/gettransaction.rs

```diff
@@ -0,0 +1,197 @@
+    pub block_hash: String,
+    pub tx: String,
+}
+
+#[derive(Clone)]
+pub struct RPCGetTransactionRequestHandler {
+    pub txid: Option<Txid>,
+}
+impl RPCGetTransactionRequestHandler {
+    pub fn new() -> Self {
+        Self { txid: None }
+    }
+}
+
+/// Decode the HTTP request
+impl HttpRequest for RPCGetTransactionRequestHandler {
+    fn verb(&self) -> &'static str {
+        "GET"
+    }
+
+    fn path_regex(&self) -> Regex {
+        Regex::new(r#"^/v3/transactions/(?P<txid>[0-9a-f]{64})$"#).unwrap()
+    }
+
+    fn metrics_identifier(&self) -> &str {
+        "/v3/transactions/:txid"
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        captures: &Captures,
+        query: Option<&str>,
+        _body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() != 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected 0-length body for GetTransaction".to_string(),
+            ));
+        }
+
+        let txid = request::get_txid(captures, "txid")?;
+        self.txid = Some(txid);
+
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCGetTransactionRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.txid = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let txid = self
+            .txid
+            .take()
+            .ok_or(NetError::SendError("`txid` no set".into()))?;
+
+        let txinfo_res =
+            node.with_node_state(|_network, _sortdb, chainstate, mempool, _rpc_args| {
+                let index_block_hashes = match chainstate.chainstate_tx_begin() {
+                    Ok((chainstate_tx, clarity_instance)) => {
+                        match chainstate_tx.get_index_block_hashes_from_txid(txid) {
+                            Ok(index_block_hashes) => index_block_hashes,
+                            Err(_) => return Err(NetError::NotFoundError),
+                        }
+                    }
+                    Err(_) => return Err(NetError::NotFoundError),
+                };
+
+                // search for the first matching tx with valid index_block_hash
+                for index_block_hash in index_block_hashes {
+                    match chainstate
+                        .nakamoto_blocks_db()
+                        .get_nakamoto_block(&index_block_hash)
+                    {
+                        Ok(nakamoto_block) => {
+                            for tx in nakamoto_block.unwrap().0.txs {
```

Comment:
We never, _ever_ `unwrap()` in network-facing code.  It's a DoS waiting to happen.  This needs to be converted into an HTTP error.